### ISSUE TITLE
markdownlint does not report all findings

### ISSUE THREAD

**Issue by clemak27** [mfussenegger/nvim-lint#70]
Hello

Currently, the markdownlinter uses the following settings:
```lua
local pattern = '(.*):(%d+):(%d+) (.*)'
local groups  = { '_', 'line', 'start_col', 'message'}
```

Let's say we have the following output from markdownlint:
```
README.md:35 MD022/blanks-around-headings/blanks-around-headers Headings should be surrounded by blank lines [Expected: 1; Actual: 0; Below] [Context: "## What's in this repo?"]
README.md:36 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "- `dotfiles`"]
README.md:47:81 MD013/line-length Line length [Expected: 80; Actual: 114]
README.md:55:81 MD013/line-length Line length [Expected: 80; Actual: 244]
```

The parser finds the Line length just fine, but fails to find the other errors. This is obviously due to the output not containing a start column.

Could a solution be to make `start_col` optional in a group and set it to zero if not found?

**Comment by mfussenegger** [comment]
Thanks for including the example output - made it easier to fix. Should now work with https://github.com/mfussenegger/nvim-lint/pull/71
